src/v/rpc/transport.cc

```diff
@@ -163,6 +164,13 @@ transport::make_response_handler(netbuf& b, const rpc::client_opts& opts) {
             _probe.request_timeout();
             _correlations.erase(it);
         }
+        /*
```

Comment:
Don't we release the units in L239? I added a test for this in the last patch .. https://github.com/redpanda-data/redpanda/blob/75fb210eaf039bf979c98fa5de9556509991220a/src/v/rpc/test/rpc_gen_cycling_test.cc#L390 

or is this an edge case (race) where a timeout kicks in before the dispatch fiber is scheduled?

Reply:
It looks like it's possible that we begin calling send after the call to `fail_outstanding_futures()` since that only calls `shutdown()`. Maybe our failure-mode calls to `fail_outstanding_futures()` should actually be calls to `stop()`?

Reply:
Also, around the other call to `_requests_queue.erase()` we explicitly move the `resource_units` and let them leave scope. Do we have to do that here?

Reply:
id we call `send` on an output stream after `shutdown` it will always throw as the underlying fd is closed. The edge case is is that the request is timed out so it returns to the caller, then the caller can continue and be destroyed. The timeout may happen before we dispatch send and then the send. Hence the units will stay in the `_requests_queue`